src/org/jitsi/impl/neomedia/rtcp/RTCPTCCPacket.java

```diff
@@ -0,0 +1,325 @@
+ * .                                                               .
+ * .                                                               .
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |         packet chunk          |  recv delta   |  recv delta   |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * .                                                               .
+ * .                                                               .
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * |           recv delta          |  recv delta   | zero padding  |
+ * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ * }</pre>
+ *
+ * @author Boris Grozev
+ */
+public class RTCPTCCPacket
+    extends RTCPFBPacket
+{
+    /**
+     * Gets a boolean indicating whether or not the RTCP packet specified in the
+     * {@link ByteArrayBuffer} that is passed as an argument is a NACK packet or
+     * not.
+     *
+     * @param baf the {@link ByteArrayBuffer}
+     * @return true if the byte array buffer holds a NACK packet, otherwise
+     * false.
+     */
+    public static boolean isTCCPacket(ByteArrayBuffer baf)
+    {
+        int rc = RTCPHeaderUtils.getReportCount(baf);
+        return isRTPFBPacket(baf) && rc == FMT;
+    }
+
+    /**
+     * The value of the "fmt" field for a transport-cc packet.
+     */
+    public static final int FMT = 15;
+
+    /**
+     * The map which contains the sequence numbers (mapped to the reception
+     * timestamp) of the packets described by this RTCP packet.
+     */
+    private Map<Integer, Long> packets = null;
+
+    /**
+     * Initializes a new <tt>NACKPacket</tt> instance.
+     * @param base
+     */
+    public RTCPTCCPacket(RTCPCompoundPacket base)
+    {
+        super(base);
+    }
+
+    /**
+     * Initializes a new {@link RTCPTCCPacket} instance with a specific "packet
+     * sender SSRC" and "media source SSRC" values, and which describes a
+     * specific set of sequence numbers.
+     * @param senderSSRC the value to use for the "packet sender SSRC" field.
+     * @param sourceSSRC the value to use for the "media source SSRC" field.
+     * @param packets the set of RTP sequence numbers and their reception
+     * timestamps which this packet is to describe.
+     * @param fbPacketCount the index of this feedback packet, to be used in the
+     * "fb pkt count" field.
+     *
+     * Note that this implementation is not optimized and might not always use
+     * the minimal possible number of bytes to describe a given set of packets.
+     * Specifically, it does take into account that sequence numbers wrap
+     * at 2^16 and fails to pack numbers close to 2^16 with those close to 0.
+     */
+    public RTCPTCCPacket(long senderSSRC, long sourceSSRC,
+                         TreeMap<Integer, Long> packets,
+                         byte fbPacketCount)
+    {
```

Comment:
lots of nice comments but still feels like a logic of magic numbers/formulae in here.  i know this is kinda the core method though, so maybe this is as good as it can be.